### DISCUSSION TITLE
How to filter by offset and limit?

### DISCUSSION THREAD

**Discussion by lz1998** [SeaQL/sea-orm#350] [Q&A] [answered]
I don't want to find by page, because the input param is offset and limit.

**Comment by billy1624** [comment, marked as answer]
Hey @lz1998,

You can do it with `Paginator`
- https://docs.rs/sea-orm/0.4.0/sea_orm/struct.Paginator.html

But I think we should have included `limit` & `offset` in our select API.

**Reply by lz1998** [reply]
What should I do if the input is `offset=3` and `count=4`?

**Reply by lz1998** [reply]
This works, but it's not convenient.
```rust
    let mut q = Entity::find();
    sea_orm::QueryTrait::query(&mut q).offset(0).limit(1);
    q.all(conn).await
```

**Reply by lz1998** [reply]
I have created a Pull Request to add offset and limit.

https://github.com/SeaQL/sea-orm/pull/351

**Reply by billy1624** [reply]
Thanks!! @lz1998 will check the PR later tonight

**Reply by lz1998** [reply]
> Thanks!! @lz1998 will check the PR later tonight

Sorry, I pushed with a wrong email and username. Could you please run the workflow to check it again?

**Reply by Shaphil** [reply]
> I have created a Pull Request to add offset and limit.
> 
> #351

Seems like your PR was merged, but I can't seem to get it to work. I currently have a similar solution as @lz1998,

```rust
let mut user_pages = User::find();
    sea_orm::QueryTrait::query(&mut user_pages)
        .offset(offset as u64)
        .limit(page_size as u64);
```

would be really nice to see an example of this chained with `find()`, like this maybe? (I tried this first and it doesn't work)
```rust
let mut user_pages = User::find()
        .order_by_asc(entity::user::Column::Id)
        .offset(offset as i64)
        .limit(page_size as i64)
        .paginate(&app_state.db, page_size as u64);
```
